addons/mail/static/src/model/model_core.js

```diff
@@ -341,16 +341,32 @@ export function patchRecordMethods(modelName, recordMethods) {
 }
 
 /**
- * @param {Object} definition The JSON definition of the model to register.
- * @param {Object} [definition.fields]
- * @param {string[]} definition.identifyingFields
- * @param {Object} [definition.lifecycleHooks]
- * @param {Object} [definition.modelGetters] Deprecated; use fields instead.
- * @param {Object} [definition.modelMethods]
- * @param {string} definition.name
- * @param {OnChanges[]} [definition.onChanges]
- * @param {Object} [definition.recordGetters] Deprecated; use fields instead.
- * @param {Object} [definition.recordMethods]
+ * @typedef {Object} LifecycleHooks
+ * @property {Function} _created Hook triggered when the model record is created.
+ * @property {Function} _willCreate Hook triggered when a model record is going to be created.
+ * @property {Function} _willDelete Hook triggered when a model record is about to be deleted.
```

Comment:
lack of consistency in words:
- `the` model/`a` model -> `a` is better (also it's probably not necessary to have the `model` word here, record should be enough?)
- is going to/is about to (no suggestion, both are fine just pick either of them)
- for the created case: it is lacking specificity about time -> "after a record is created"?

Reply:
Do you need Function with uppercase to make it work? Our guideline so far is to have it lowercase.

Reply:
It can work with function.